main.go

```diff
@@ -137,6 +139,96 @@ func main() {
 		rootCmd.AddCommand(cmdDecrypt)
 	}
 
+	// Exec command
+	{
+		var secretaryURL, appID, appVersion, taskID string
+		var configKeyFile, masterKeyFile, deployKeyFile, serviceKeyFile string
+		cmdExec := &cobra.Command{
+			Use:   "exec",
+			Short: "execute program, decrypting environment variables and arguments",
+			PreRun: func(cmd *cobra.Command, args []string) {
+				if len(args) == 0 {
+					fmt.Print("Error: Must supply at least one command\n\n")
+					cmd.Usage()
+					os.Exit(1)
+				}
+			},
+			Run: func(cmd *cobra.Command, args []string) {
+				var crypto DecryptionStrategy
+
+				if len(secretaryURL) > 0 {
+					deployKey := requireKey("deploy private", deployKeyFile, "DEPLOY_PRIVATE_KEY", "./keys/master-private-key.pem")
+					masterKey := requireKey("master public", masterKeyFile, "MASTER_PUBLIC_KEY", "./keys/master-public-key.pem")
+					serviceKey := findKey(serviceKeyFile, "SERVICE_PRIVATE_KEY")
+					crypto = newDaemonDecryptionStrategy(
+						secretaryURL, appID, appVersion, taskID,
+						masterKey, deployKey, serviceKey)
+				} else {
+					// Send ENC[KMS,..] and ENC[NACL,...] to separate decryptors
+					composite := newCompositeDecryptionStrategy()
+					composite.Add("KMS", newKmsDecryptionStrategy(newKmsClient()))
+
+					deployKey := findKey("deploy private", deployKeyFile, "DEPLOY_PRIVATE_KEY", "./keys/master-private-key.pem")
+					configKey := findKey("config public", configKeyFile, "CONFIG_PUBLIC_KEY", "./keys/config-public-key.pem")
+					if deployKey != nil && configKey != nil {
+						composite.Add("NACL", newKeyDecryptionStrategy(configKey, deployKey))
+					}
+
+					crypto = composite
+				}
+
+				newCmd, newArgs, newEnvs, err := createExecArgs(args, os.Environ(), crypto)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "%v", err)
+					os.Exit(1)
+				}
+
+				if err := syscall.Exec(newCmd, newArgs, newEnvs); err != nil {
+					fmt.Fprintf(os.Stderr, "Error when executing command \"%s\": %v\n", newCmd, err)
+				}
+			},
+		}
+
+		cmdExec.SetUsageTemplate(`Usage:{{if .Runnable}}
```

Comment:
Why is this needed? None of the other commands need to specify a convoluted template like this?